### PR TITLE
Linter: Update implementation code

### PR DESCRIPTION
**Pull request by JoshVanL**
Update Dapr implementation code to adhere to new linter rules. All test files have been purposefully omitted to ensure that the implementation code is still functionally correct. A follow-up commit will contain the update to all test code, and removal of the "only new changes linted" from the github linter workflow.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -16,28 +16,29 @@ run:
   build-tags:
     - unit
```
Comment (mikeee) [inline .golangci.yml:17]:
```suggestion
    - unit
    - allcomponents
    - subtlecrypto
```

Missing build tags - as per workflow

File: .golangci.yml [hunk 2]
```diff
@@ -86,9 +84,8 @@ linters-settings:
     # enable or disable analyzers by name
     enable:
       - atomicalign
+      - shaddow
```
Comment (mikeee) [inline .golangci.yml:87]:
```suggestion
      - shadow
```
sp.

File: pkg/api/errors/pubsub.go [hunk 3]
```diff
@@ -150,7 +150,7 @@ func (p *PubSubMetadataError) TopicEmpty() error {
 	return p.build(
 		codes.InvalidArgument,
 		http.StatusNotFound,
-		fmt.Sprintf("topic is empty in pubsub %s", p.p.name),
+		"topic is empty in pubsub "+p.p.name,
```
Comment (mikeee) [inline pkg/api/errors/pubsub.go:153]:
```suggestion
		"topic is empty in pubsub " + p.p.name,
```
I'm surprised gofmt didn't complain about this

**Review by mikeee — CHANGES_REQUESTED** [review]

**Review by elena-kolevska — APPROVED** [review]
LGTM, provided that Mike's suggestions are accepted

File: .golangci.yml [hunk 4]
```diff
@@ -16,28 +16,29 @@ run:
...
-  # autogenerated files. If it's not please let us know.
-  # skip-files:
-  #  - ".*\\.my\\.go$"
-  #  - lib/bad.go
-  
 # output configuration options
 output:
   # colored-line-number|line-number|json|tab|checkstyle, default is "colored-line-number"
-  format: tab
+  formats: tab
```
Comment (elena-kolevska) [inline .golangci.yml:41]:
```suggestion
  formats:
    - format: tab
```

File: .golangci.yml [hunk 5]
```diff
@@ -60,7 +61,7 @@ linters-settings:
     # [deprecated] comma-separated list of pairs of the form pkg:regex
     # the regex is used to ignore names within pkg. (default "fmt:.*").
     # see https://github.com/kisielk/errcheck#the-deprecated-method for details
-    ignore: fmt:.*,io/ioutil:^Read.*
+    exclude-functions: fmt:.*,io/ioutil:^Read.*
```
Comment (elena-kolevska) [inline .golangci.yml:64]:
```suggestion
    exclude-functions:
      - fmt:.*
      - io/ioutil:^Read.*
```

File: .golangci.yml [hunk 6]
```diff
@@ -86,9 +84,8 @@ linters-settings:
     # enable or disable analyzers by name
     enable:
       - atomicalign
+      - shaddow
     enable-all: false
-    disable:
-      - shadow
     disable-all: false
   golint:
```
Comment (elena-kolevska) [inline .golangci.yml:90]: The `golint` linter was removed, so it should be removed from settings. Same with `maligned`

**Review by elena-kolevska — CHANGES_REQUESTED** [review]
The typo `shaddow` made me think there could be other errors on which the tool failed silently, so I did a quick search and found out about the `golangci-lint config verify` command. It brought a few additional errors we need to fix. I put a few of them as suggestions, and here are two more I can't add as comments in code:

Under `linter-settings`:
```
# `check-exported: false` is not  valid anymore, I suggest: 
unused:
    # Mark all exported fields as used.
    exported-fields-are-used: false
```    
```
# We used to have `allow-case-traling-whitespace: true` which is not allowed anymore. We can either delete the config completely, or force a new line as so:
wsl:
  ...
      # If the number of lines in a case block is equal to or lager than this number,
    # the case *must* end white a newline.
    # https://github.com/bombsimon/wsl/blob/master/doc/configuration.md#force-case-trailing-whitespace
    # Default: 0
    force-case-trailing-whitespace: 1
```

**Comment by JoshVanL** [comment]
Thanks @elena-kolevska @mikeee, please take another look!

**Review by elena-kolevska — APPROVED** [review]

**Review by mikeee — APPROVED** [review]
Lgtm

File: pkg/api/errors/pubsub.go (hunk 3 above)
Comment (elena-kolevska) [reply pkg/api/errors/pubsub.go:153]: TIL: gofmt doesn't force spaces around operators if the expression is an argument of a function with more than one argument 🤷‍♀️ 
Ref: https://github.com/golang/go/issues/34426, https://github.com/golang/go/issues/12105

**Review by cicoyle — APPROVED** [review]

**Pull request merged**